### PR TITLE
task: adding local deploy targets

### PR DESCRIPTION
**Pull request by o11yguru**
### Any background context you want to provide for reviewers?
These are the relevant steps for a local engine install. 

### What is the focus or goal of the code changes?
Provide a local approach to test the tech before deploying to a CSP like AWS.

### Where should the reviewer start?
`README.md` ln. 29

### Does this add new dependencies? Is there an installation or build procedure? 
Yes, there are 4 new build procedures. 
1.`make -f ./make/Makefile-local-recipies create-mdai`              deploy a local cluster from scratch
2. `make -f ./make/Makefile-local-recipies delete-mdai`             deletes mdai cluster deployed locally and all artifacts associated
3. `make -f ./make/Makefile-local-recipies delete-mdai-all`             deletes mdai cluster deployed locally and all artifacts associated, plus helm charts
4. `make -f ./make/Makefile-local-recipies update-mdai-collector`              updates mdai the collector to the latest configuration

### How should this be manually tested?
Run each of the commands in their correct use as called out above. 

### Screenshots (if applicable)
n/a 

### Optional: What gif best describes this PR or how it makes you feel?
it's happening

### CODE REVIEW

File: make/Makefile-local-install [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# ~~~~~~~~~~~ INSTALL DEPENDENCY SCRIPT ~~~~~~~~~~~~~~
+
+# Install system requirements for MDAI
+# .SILENT: mdai-system-reqs
+.PHONY: mdai-system-reqs
+mdai-system-reqs:
+  @echo "🟢 Start mdai-system-reqs..."
+  @go version || brew list go || brew install go \
+  kubectl version || brew list kubectl || brew install kubectl \
+  npm -v || brew list npm || brew install npm \
```
Comment (tsloughter) [inline make/Makefile-local-install:14]: Why is `npm` needed? I don't see it used, but maybe some other command is using it?

File: make/Makefile-local-install [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# ~~~~~~~~~~~ INSTALL DEPENDENCY SCRIPT ~~~~~~~~~~~~~~
+
+# Install system requirements for MDAI
+# .SILENT: mdai-system-reqs
+.PHONY: mdai-system-reqs
+mdai-system-reqs:
+  @echo "🟢 Start mdai-system-reqs..."
+  @go version || brew list go || brew install go \
```
Comment (tsloughter) [inline make/Makefile-local-install:12]: Is go needed for like `brew install kubectl`?

File: README.md [hunk 3]
```diff
@@ -24,3 +26,15 @@
 * `cdk diff`        compare deployed stack with current state
 * `cdk synth`       emits the synthesized CloudFormation template
 
+## Local Deployment commands
+
+* `make -f ./make/Makefile-local-recipies create-mdai`              deploy a local cluster from scratch
+* `make -f ./make/Makefile-local-recipies delete-mdai`             deletes mdai cluster deployed locally and all artifacts associated
+* `make -f ./make/Makefile-local-recipies delete-mdai-all`             deletes mdai cluster deployed locally and all artifacts associated, plus helm charts
+* `make -f ./make/Makefile-local-recipies update-mdai-collector`              updates mdai the collector to the latest configuration
```
Comment (arcanez) [inline README.md:34]:
```suggestion
* `make -f ./make/Makefile-local-recipes create-mdai`              deploy a local cluster from scratch
* `make -f ./make/Makefile-local-recipes delete-mdai`              deletes mdai cluster deployed locally and all artifacts associated
* `make -f ./make/Makefile-local-recipes delete-mdai-all`         deletes mdai cluster deployed locally and all artifacts associated, plus helm charts
* `make -f ./make/Makefile-local-recipes update-mdai-collector`   updates mdai the collector to the latest configuration
```

File: jobs/cronjob-logs.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ kind: CronJob
 metadata:
   name: telemetrygen-logs
 spec:
-  # Change this to the frequency you desire -- currently it's set to 5 minutes
+  # Change this to the frequency you desire -- currently it's set to 1 minutes
```
Comment (arcanez) [inline jobs/cronjob-logs.yaml:6]:
```suggestion
  # Change this to the frequency you desire -- currently it's set to 1 minute
```

File: jobs/cronjob-metrics.yaml [hunk 5]
```diff
@@ -3,9 +3,9 @@ kind: CronJob
 metadata:
   name: telemetrygen-metrics
 spec:
+  # Change this to the frequency you desire -- currently it's set to 1 minutes
```
Comment (arcanez) [inline jobs/cronjob-metrics.yaml:6]:
```suggestion
  # Change this to the frequency you desire -- currently it's set to 1 minute
```

File: make/Makefile-local-recipies [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# ~~~~~~~~~~~~~~~~~ RECIPIES ~~~~~~~~~~~~~~~~~
```
Comment (arcanez) [inline make/Makefile-local-recipies:1]:
```suggestion
# ~~~~~~~~~~~~~~~~~ RECIPES ~~~~~~~~~~~~~~~~~
```

File: jobs/cronjob-metrics.yaml [hunk 7]
```diff
@@ -3,9 +3,9 @@ kind: CronJob
...
   schedule: "*/1 * * * *"
   concurrencyPolicy: Forbid
-  successfulJobsHistoryLimit: 1
```
Comment (arcanez) [inline jobs/cronjob-metrics.yaml:8]: curious why this was removed

**Review by arcanez — CHANGES_REQUESTED** [review]
rename `make/Makefile-local-recipies` to `make/Makefile-local-recipes`

File: make/Makefile-local-install (hunk 1 above)
Comment (arcanez) [reply make/Makefile-local-install:14]: good call out, `npm` is needed for AWS (`cdk`) but not local install

File: make/Makefile-local-install [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  @echo "🟢 Start mdai-wait-for-pods..."
+  kubectl -n default wait --for condition=ready pod -l app.kubernetes.io/name=opentelemetry-operator
+  kubectl -n default wait --for condition=ready pod -l app.kubernetes.io/component=controller-manager
+  @echo "✅ Complete mdai-wait-for-pods!"
+
+# deploy collector config
+.SILENT: mdai-deploy-config
+.PHONY: mdai-deploy-config
+mdai-deploy-config:
+  kubectl apply -f ./templates/otel-collector.yaml
```
Comment (tsloughter) [inline make/Makefile-local-install:73]: Why does this have to be applied here and not part of the deploy of the helm chart?

Comment (arcanez) [reply make/Makefile-local-install:73]: this allows multiple applications of the collector config without redploying the helm chart (even if the helm chart deploy is a noop)

Comment (tsloughter) [reply make/Makefile-local-install:73]: Aahh!

File: make/Makefile-local-install [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+	@helm repo list | grep mydecisive || helm repo add mydecisive https://decisiveai.github.io/mdai-helm-charts
+	@echo "✅ Complete mdai-add-helm-charts!"
+
+# Install helm charts to cluster
+.SILENT: mdai-install-helm-charts
+.PHONY: mdai-install-helm-charts
+mdai-install-helm-charts:
+	@echo "🟢 Start mdai-install-helm-charts..."
+	helm repo update
+	helm upgrade -f ./templates/prometheus-values.yaml prometheus prometheus-community/prometheus --install --wait
```
Comment (weird-bytecode) [inline make/Makefile-local-install:38]: I fixed a file path here, changing `prometheus_values.yaml` to `prometheus-values.yaml` when I did my tabs fixes. However I think maybe the values in that file actually might break the metrics scraping 😬

**Review by arcanez — APPROVED** [review]

**Review by arcanez — APPROVED** [review]

**Pull request merged**